src/waveform/vsyncthread.cpp

```diff
@@ -32,8 +36,8 @@ void VSyncThread::run() {
     m_timer.start();
 
     //qDebug() << "VSyncThread::run()";
-    while (m_bDoRendering) {
-        if (m_vSyncMode == ST_FREE) {
+    if (m_vSyncMode == ST_FREE) {
```

Comment:
The code in the different if-else arms should be extracted into separate functions `render_free()`/`render_pll()`/`render_timer()`. Use a debug assertion in each function to verify that `m_vSyncMode` matches.

Using `switch(m_vSyncMode)` for the different cases would be more idiomatic.

Reply:
I agree but that is part of https://github.com/mixxxdj/mixxx/pull/12469 , I will change it there.

(There might be a better way to create a PR on top of other PR, to avoid this confusion)

Reply:
done